tensorflow/python/layers/layers.py

```diff
@@ -31,6 +31,7 @@
 @@average_pooling3d
 @@max_pooling3d
 @@batch_normalization
+@@maxout
```

Comment:
Instead of exposing the symbol here, how about exposing it from contrib.layers (leaving the code where it is)? This way we can work out if any APIs need to change before committing to backward compatibility forever.

Reply:
Hmmm, I don't understand why this symbol is not suitable here

Reply:
Adding the symbol here will add tf.layers.maxout (but not tf.layers.Maxout, the object-based version). In tensorflow, since 1.0, we are promising full backward compatibility on all APIs which are not in tf.contrib. Since maxout was just added I don't think it deserves full backward compatibility promises yet. So I prefer that the symbol is added to tf.contrib.layers.{Maxout,maxout}

Reply:
Ok, understood.
Shall I keep the code in the same place and just import the symbol in the contrib or I should move everything?

Reply:
Yes, the code is in the right place.

Reply:
Done, but I still don't know how to make the sanity checks green.
The unit tests and pylint works in my workstation

Shall I squash my commits everything before merge?